sdks/java/core/src/main/java/org/apache/beam/sdk/testing/SourceTestUtils.java

```diff
@@ -673,4 +680,93 @@ public SplitAtFractionResult(int numPrimaryItems, int numResidualItems) {
         numItemsToReadBeforeSplitting, fraction, options);
     return (res.numResidualItems > 0);
   }
+
+  /**
+   * Returns an equivalent unsplittable {@code BoundedSource<T>}.
+   */
+  public static <T> BoundedSource<T> toUnsplittableSource(BoundedSource<T> boundedSource) {
+    return new UnsplittableSource<>(boundedSource);
+  }
+
+  private static class UnsplittableSource<T> extends BoundedSource<T> {
+
+    private final BoundedSource<T> boundedSource;
+
+    private  UnsplittableSource(BoundedSource<T> boundedSource) {
+      this.boundedSource = checkNotNull(boundedSource, "boundedSource");
+    }
+
+    @Override
+    public List<? extends BoundedSource<T>> splitIntoBundles(
```

Comment:
implement the rest of the methods on BoundedSource and BoundedReader. (there are some that are not abstract, but have default implementations).